app/models/course.rb

```diff
@@ -4,7 +4,7 @@ class Course < ApplicationRecord
   has_many :application_choices, through: :course_options
   has_many :sites, through: :course_options
   has_many :course_subjects
-  has_many :subjects, through: :course_subjects
+  has_many :subjects, -> { order('id ASC') }, through: :course_subjects
```

Comment:
I agree with attempting to avoid a default scope as to avoid any repercussions in the rest of the system. Could we instead pass in a course or application choice to `determine_dominant_course_subject_for_report` and calculate the `subject_names_and_codes` inside that method and determine the scope there? Appreciate this might be a bit more involved, let me know your thoughts